File: app/src/main/java/com/rr/tvstarterkit/application/ContextProvider.java
package com.rr.tvstarterkit.application;

import android.content.Context;

public class ContextProvider {

    private Context applicationContext;

    private static ContextProvider mContextProvider;

    private ContextProvider() {
    }

    public static ContextProvider getContextProvider() {
        if (mContextProvider == null) {
            synchronized (ContextProvider.class) {
                mContextProvider = new ContextProvider();
            }
        }

        return mContextProvider;
    }

    public Context provideContext() {
        return applicationContext;
    }

    public void setApplicationContext(Context context) {
        applicationContext = context;
    }

    public void destroyContext() {
        applicationContext = null;
    }
}

File: app/src/main/java/com/rr/tvstarterkit/application/ContextProvider.kt
package com.rr.tvstarterkit.application

import android.content.Context
import com.rr.tvstarterkit.application.ContextProvider

class ContextProvider private constructor() {
    private var applicationContext: Context? = null
    fun provideContext(): Context? {
        return applicationContext
    }

    fun setApplicationContext(context: Context?) {
        applicationContext = context
    }

    fun destroyContext() {
        applicationContext = null
    }

    companion object {
        private var mContextProvider: ContextProvider? = null
        @JvmStatic
        val contextProvider: ContextProvider?
            get() {
                if (mContextProvider == null) {
                    synchronized(ContextProvider::class.java) {
                        mContextProvider = ContextProvider()
                    }
                }
                return mContextProvider
            }
    }
}